### PR TITLE
Returnhook

### PR DESCRIPTION
**Pull request by wizage**
*Issue #, if available:* #748 

*Description of changes:* Adds functionality for callbacks to be implemented by any plugin that amplify supports (both 3rd party and 1st party plugins can be supported with this). This just builds on top of the callback for frontend development and adds support for the categories.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-hosting/lib/S3AndCloudFront/template.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
                         }      
                     ]
                 },
-                "AccessControl": "PublicRead",
```
Comment (kaustavghosh06) [inline packages/amplify-category-hosting/lib/S3AndCloudFront/template.json]: Don't think that this is a part of this PR? Make sure to rebase

**Comment by wizage** [comment]
@kaustavghosh06 Fixed. Sorry about that.

**Comment by derrick3192** [comment]
How does one access this hook when creating a plugin? Would be great to include it in here [Plugins](https://aws-amplify.github.io/docs/cli/plugins)

**Comment by wizage** [comment]
@derrick3192 We are working on updating the docs but to use it you just need to add this code to your index.js file.
```
async function onAmplifyCategoryOutputChange(context) {
  //callback code
}
```

**Comment by derrick3192** [comment]
Awesome! Appreciate the replly cheers.

**Pull request merged**